pkg/control_room/control_center.go

```diff
@@ -0,0 +1,99 @@
+package control_room
+
+import (
+	"container/heap"
+
+	"github.com/TNSEngineerEdition/WailsClient/pkg/city"
+	"github.com/umahmood/haversine"
+)
+
+type ControlCenter struct {
+	city         *city.City
+	cachedRoutes map[[2]uint64][]*city.GraphNode
+}
+
+func CreateControlCenter(cityData *city.City) ControlCenter {
```

Comment:
Probably this is the better name for this argument
```suggestion
func CreateControlCenter(city *city.City) ControlCenter {
```

Reply:
But if we name this argument `city` then this argument shadows city package and occurs error: `city.GraphNode is not a typecompiler[NotAType](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#NotAType)`. So maybe I will rename this as `cty`? What do you think?

Reply:
Good point, maybe `cityPointer` is a better name in this case?